typescript/infra/config/routingIsm.ts

```diff
@@ -34,81 +39,102 @@ const owners = {
+//                 |              |
+//                 |              |
+//                 v              v
+//            Merkle Root    Message ID
 export const routingIsm = (
   environment: DeployEnvironment,
   local: ChainName,
-  type: MultisigIsmConfig['type'],
   context: Contexts,
 ): RoutingIsmConfig => {
-  const defaultMultisigIsmConfigs = multisigIsms(
-    environment,
-    local,
-    type,
-    context,
+  const aggregationIsms: ChainMap<AggregationIsmConfig> = chains[
+    environment
+  ].reduce(
+    (acc, chain) => ({
+      ...acc,
+      [chain]: aggregationIsm(chain, context),
+    }),
+    {},
   );
+
   return {
     type: ModuleType.ROUTING,
-    domains: defaultMultisigIsmConfigs,
+    domains: aggregationIsms,
     owner: owners[environment][local],
   };
 };
 
-// mainnet cache
-const aggregationIsmAddresses: Record<string, string> = {
-  [Chains.arbitrum]: '0x7995D00bdDb146334d6568b627bcd2a7DdA3B005',
-  [Chains.avalanche]: '0xF6bF41939ebA2363A6e311E886Ed4a5ab3dc1F5D',
-  [Chains.bsc]: '0x294F19d5fe29646f8E2cA4A71b6B18b78db10F9f',
-  [Chains.celo]: '0x656bF500F0E2EE55F26dF3bc69b44c6eA84dd065',
-  [Chains.ethereum]: '0xe39eA548F36d1c3DA9b871Badd11345f836a290A',
-  [Chains.gnosis]: '0xD0Ec4de35069520CD17522281D36DD299525d85f',
-  [Chains.moonbeam]: '0x04100049AC8e279C85E895d48aab1E188152e939',
-  [Chains.optimism]: '0x99663d142576204284b91e96d39771db94eD5188',
-  [Chains.polygon]: '0x0673cc1cc5eb80816E0d0E2dA5FE10053Da97943',
-};
-
-/// 1/2 Aggregation => Routing => Multisig ISM
+// Aggregation (1/2)
+// |              |
+// |              |
+// v              v
+// Merkle Root    Message ID
 export const aggregationIsm = (
-  environment: DeployEnvironment,
-  local: ChainName,
+  remote: ChainName,
   context: Contexts,
-): AggregationIsmConfig | Address => {
-  if (environment === 'mainnet2') {
-    return aggregationIsmAddresses[local];
-  }
-
-  const config: AggregationIsmConfig = {
+): AggregationIsmConfig => {
+  return {
     type: ModuleType.AGGREGATION,
     modules: [
-      // ORDERING MATTERS
-      routingIsm(environment, local, ModuleType.MERKLE_ROOT_MULTISIG, context),
-      routingIsm(environment, local, ModuleType.MESSAGE_ID_MULTISIG, context),
+      // Ordering matters to preserve determinism
+      multisigIsm(remote, ModuleType.MERKLE_ROOT_MULTISIG, context),
+      multisigIsm(remote, ModuleType.MESSAGE_ID_MULTISIG, context),
     ],
     threshold: 1,
   };
-  return config;
 };
 
+export const multisigIsm = (
+  remote: ChainName,
+  type: MultisigIsmConfig['type'],
+  context: Contexts,
+): MultisigIsmConfig => {
+  const configs =
+    context === Contexts.ReleaseCandidate
+      ? rcMultisigIsmConfigs
+      : defaultMultisigIsmConfigs;
+
+  return {
+    ...configs[remote],
+    type,
+  };
+};
+
+export const multisigIsms = (
+  env: DeployEnvironment,
+  local: ChainName,
+  type: MultisigIsmConfig['type'],
+  context: Contexts,
+): ChainMap<MultisigIsmConfig> =>
+  objMap(
```

Comment:
When I think about the code in infra that may have trouble finding a home in the CLI, all this utils-like stuff for creating/transforming configs comes to mind. 
Not a criticism of the code here, just thinking out loud

Reply:
Yeah this is a great callout imo

I think we'll probably want to maintain the ability to programmatically construct these things if possible. I have some ideas but we should chat